src/win32/thread_primitives.cpp

```diff
@@ -93,8 +94,24 @@ ticks_type WINAPI get_tick_count_init()
     boost::winapi::HMODULE_ hKernel32 = boost::winapi::GetModuleHandleW(L"kernel32.dll");
     if (hKernel32)
     {
+      // GetProcAddress returns a pointer to some function. It can return
+      // pointers to different functions, so it has to return something that is
+      // suitable to store any pointer to function. Microsoft chose FARPROC,
+      // which is int (WINAPI *)() on 32-bit Windows. The user is supposed to
+      // know the signature of the function he requests and perform a cast
+      // (which is a nop on this platform). The result is a pointer to function
+      // with the required signature, which is bitwise equal to what
+      // GetProcAddress returned.
+      // However, gcc warns about that.
+#if BOOST_COMP_GNUC
+#pragma GCC diagnostic push
```

Comment:
Pragma diagnostic push/pop appeared only since gcc 4.6. You should check for the compiler version.

Also, you don't need Boost.Predef to detect gcc. You can use `BOOST_GCC` from Boost.Config. Also, it is quite possible that clang also emits a similar warning, you can check for it with `BOOST_CLANG`.

Reply:
I cannot check for clang as I do not cross-compile to Windows with clang but I will use BOOST_GCC.